tservice/types/src/seal.rs

```diff
@@ -69,6 +77,55 @@ impl<'a> TryFrom<&'a [u8]> for SealedData<'a> {
     }
 }
 
+#[cfg(feature = "alloc")]
+impl TryFrom<Vec<u8>> for SealedData<Vec<u8>> {
+    type Error = FfiError;
+    fn try_from(bytes: Vec<u8>) -> Result<Self> {
+        let offset = mem::size_of::<sgx_sealed_data_t>() - mem::size_of::<sgx_aes_gcm_data_t>();
+        let aes_gcm_bytes = bytes.get(offset..).ok_or(FfiError::InvalidInputLength)?;
+        AesGcmData::try_from(aes_gcm_bytes)?;
+        Ok(Self { bytes })
+    }
+}
+
+/// Unsealed Data
+///
+/// A plain old data type (POD) of the component pieces of the data stored in
+/// [`sgx_sealed_data_t`]
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+pub struct UnsealedData<T> {
+    /// The data to be encrypted/sealed
+    pub data: T,
+
+    /// The MAC text that will not be encrypted
+    pub mac: Option<T>,
+}
```

Comment:
This is going to sound weird at first blush, but this should be in the `tservice` crate, not `tservice-types`. I say so because having data that is meant to be sealed be accessible outside the enclave is always an error

Reply:
I think that makes sense I was just blanket throwing all types into the types crates, but as you say `unsealed` data shouldn't leak outside of the enclave, once it does it's sealing it is a false feeling of security.

Reply:
done now